catkin_home/src/main_engine/src/action_manager.py

```diff
@@ -70,7 +75,16 @@ def new_action_request_callback(self, msg):
         self.action_queue.append(new_action)
         print("new action added to the queue")
      
-
+    def createAction(self,msg):
+       # TODO: See if dynamically create classes based on the string
+       # target_class = getattr(actions, msg.intent)
+        if(msg.intent == "go_to"):
```

Comment:
We could change this to a switch for future development (and efficiency); There's gonna be more than 4 intents.
https://stackoverflow.com/questions/767821/is-else-if-faster-than-switch-case

Reply:
There is no switch statement in python, that's why I added the TODO to use a dict or something else to create the classes dynamically.